rest/route/middleware.go

```diff
@@ -423,6 +423,48 @@ func (m *podAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, n
 	next(rw, r)
 }
 
+type alertmanagerMiddleware struct{}
+
+// NewAlertmanagerMiddleware returns a middleware that verifies the request
+// is coming from Evergreen's configured Alertmanager Kanopy webhook.
+func NewAlertmanagerMiddleware() gimlet.Middleware {
+	return &alertmanagerMiddleware{}
+}
+
+func (m *alertmanagerMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	username, password, ok := r.BasicAuth()
```

Comment:
Since this isn't how we usually do auth to  Evergreen, can we call out in a comment why this has to use basic auth instead of the usual auth username/API key headers?